message_ix/tests/test_feature_duration_time.py

```diff
@@ -0,0 +1,255 @@
+# -*- coding: utf-8 -*-
+"""
+This test ensures that the parameter "duration_time" is specified correctly,
+and that the GAMS formulation checks pass.
+Several possible combinations of temporal levels, e.g., year, seasons,
+months; and different number of time slices at each level are tested.
+
+"""
+
+from itertools import product
+
+from message_ix import Scenario
+
+
+# A function for generating a simple model with sub-annual time slices
+def model_generator(
+    test_mp,
+    comment,
+    tec_time,
+    demand_time,
+    time_steps,
+    com_dict,
+    yr=2020,
+):
+    """
+
+    Generates a simple model with a few technologies, and a flexible number of
+    time slices.
+
+    Parameters
+    ----------
+    comment : string
+        Annotation for saving different scenarios and comparing their results.
+    tec_time : dict
+        A dictionary for mapping a technology to its input/output temporal levels.
+    demand_time : dict
+        A dictionary for mapping the total "demand" specified at a temporal level.
+        e.g., {"season": 100}, means the sum of demand in all "time"
+        slices at the level of "season" must be 100.
```

Comment:
Should we remove also this example or is it fine to leave it here?

Reply:
Yes, we can remove this. I resolved the remaining issues. Thanks, @LauWien.